docs/coverage.md

```diff
@@ -0,0 +1,82 @@
+# Coverage
+
+Syzkaller data exported by kcov is used to determine coverage. Kcov is exporting first PC value of each executed code block (see https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/Documentation/dev-tools/kcov.rst). Tools from binutils (objdump, nm, addr2line and readelf) are used to map this data to lines and functions in the source code.
+
+## Binutils
+
+Note that if you are fuzzing in cross-arch environment you need to provide correct Binutils cross-tools to Syzkaller before starting Syz-manager:
+
+```
+mkdir -p ~/bin/mips64le
+ln -s `which mips64el-linux-gnuabi64-addr2line` ~/bin/mips64le/addr2line
+ln -s `which mips64el-linux-gnuabi64-nm` ~/bin/mips64le/nm
+ln -s `which mips64el-linux-gnuabi64-objdump` ~/bin/mips64le/objdump
+ln -s `which mips64el-linux-gnuabi64-readelf` ~/bin/mips64le/readelf
+export PATH=~/bin/mips64le:$PATH
+```
+
+### Objdump
+Objdump is used to parse PC value of each call to __sanitizer_cov_trace_pc in the kernel image. These PC values are representing all code that is built into kernel image. PC values exported by kcov are compared against these to determine coverage.
+
+### Nm
+Nm is used to parse address and size of each function in the kernel image. This information is used to map coverage data to functions. This is needed to find out whether certain functions are called at all.
+
+### Addr2line
+Addr2line is used for mapping PC values exported by kcov and parsed by objdump to source code files and lines.
+
+### Readelf
+Readelf is used to detect virtual memory offset. Executor is truncating PC values into uint32 before sending them to Syz-manager and Syz-manager has to detect the offset.
+
+## Web Interface
+
+When clicking on "cover" link you get view showing each directory located in your kernel build directory. It's showing either percentage number or "---". "---" indicates code in that directory is not instrumented by kcov or there are not yet coverage in that directory.
+
+Directory can be clicked and you get view on files and possible subdirectories. On each source code file there is again either "---" or coverage percentage.
+
+If you click on any C files you will get source code view. There is certain coloring used in the source code view. Color definitions can be found in [coverTemplate](/pkg/cover/report.go#L504). Coloring is described below. 
+
+### Covered RGB(0, 0, 0)
+
+All PC values associated to that line are covered. There is number on the left side indicating how many programs have triggered executing the PC values assocaciated to this line. You can click on that number and it will open last executed program. Example below shows how single line which is fully covered is shown
+
+![Code line is fully covered](covered.png?raw=true)
+
+### Both RGB(255, 100, 0)
```

Comment:
Something like `Both: orange (#ff6400)` here and in other places.

Reply:
Fixed